gobblin-core/src/main/java/gobblin/writer/EnvelopeDataWriterBuilder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.writer;
+
+import gobblin.configuration.State;
+import gobblin.writer.partitioner.SchemaBasedWriterPartitioner;
+import org.apache.avro.Schema;
+import org.apache.hadoop.fs.Path;
+
+
+/**
+ * A {@link DataWriterBuilder} for use with {@link gobblin.converter.EnvelopeSchemaConverter}
+ */
+public class EnvelopeDataWriterBuilder extends AvroDataWriterBuilder {
```

Comment:
Can you make this class name more generic? Its implementation does not imply `Envelope`.